### PR TITLE
Adds startsWith and endsWith validation rules

### PR DESCRIPTION
**Pull request by andrew-boyd**
- adds support and tests for startsWith and endsWith validation rules
- re-orders rules and rules tests files to list rules alphabetically
- converts snake_case validation rules to camelCase function calls internally

### CODE REVIEW

**Review by justin-schroeder — CHANGES_REQUESTED** [review]
Nice work! Added a couple quick requests

File: src/locales/en.js [hunk 1]
```diff
@@ -89,6 +89,16 @@ export default {
     return `“${value}” is not a valid email address.`
   },
 
+  /**
+   * Ends with specified value
+   */
+  endsWith: function ({ name, value }) {
+    if (!value) {
```
[the remark this answers is not in the document]
Comment (andrew-boyd) [reply src/locales/en.js:96]: I think this change has already been made. If the value is empty the message is now a more explicit "This field doesn’t end with a valid value."

File: src/locales/en.js [hunk 2]
```diff
@@ -162,6 +172,16 @@ export default {
     return `${s(name)} is required.`
   },
 
+  /**
+   * Starts with specified value
+   */
+  startsWith: function ({ name, value }) {
```
[the remark this answers is not in the document]
Comment (andrew-boyd) [reply src/locales/en.js:178]: same as above

File: test/rules.test.js [hunk 3]
```diff
@@ -291,11 +206,88 @@ describe('email', () => {
...
 
-  it('fails string without tld', async () => expect(await rules.email({ value: '12345@localhost' })).toBe(false))
-
   it('fails string without invalid name', async () => expect(await rules.email({ value: '1*(123)2345@localhost' })).toBe(false))
 })
 
+/**
+ * Checks if value ends with a one of the specified Strings.
+ */
+describe('endsWith', () => {
```
[the remark this answers is not in the document]
Comment (andrew-boyd) [reply test/rules.test.js:215]: good question. I would expect this rule would be invalid on checkboxes, radios, selects, images, etc. 

Would your preference be for it to pass or fail when applied to a non-text type input?

File: src/libs/utils.js [hunk 4]
```diff
@@ -71,6 +71,16 @@ export function shallowEqualObjects (objA, objB) {
   return true
 }
 
+/**
+ * Given a string, convert snake_case to camelCase
+ * @param {String} string
+ */
+export function snakeCaseToCamelCase (string) {
```
[the remark this answers is not in the document]
Comment (andrew-boyd) [reply src/libs/utils.js:78]: Good call, will do.

[the remark this answers is not in the document]
Comment (andrew-boyd) [reply src/libs/utils.js:78]: Tests added in https://github.com/wearebraid/vue-formulate/pull/9/commits/284a49287b53fb9734a4c4abf9d0b3dea717640b

**Review by justin-schroeder — APPROVED** [review]

**Comment by andrew-boyd** [comment]
@justin-schroeder i still need to add tests for the new validation rules with non-string arguments that either pass or fail as desired. 

Did you have opinion on if passing an array or object should fail, or be ignored (pass)?

**Comment by justin-schroeder** [comment]
Generally my opinion is that any non-passing test should fail by default, this might not be upheld perfectly in the existing rules, but the question the rules should be asking is "does it pass?" – rather than "does it fail?".

**Comment by andrew-boyd** [comment]
Makes sense. I’ll get the edge cases taken care of with tests that ensure they are failing.
